pallets/manta-pay/src/rpc.rs

```diff
@@ -19,25 +19,30 @@
 use crate::{runtime::PullLedgerDiffApi, PullResponse};
 use alloc::sync::Arc;
 use core::marker::PhantomData;
-use jsonrpc_core::{Error, ErrorCode, Result};
-use jsonrpc_derive::rpc;
+use jsonrpsee::{
+    core::{async_trait, RpcResult},
+    proc_macros::rpc,
+    types::error::{CallError, ErrorObject},
+};
 use manta_pay::signer::Checkpoint;
 use sp_api::ProvideRuntimeApi;
 use sp_blockchain::HeaderBackend;
 use sp_runtime::{generic::BlockId, traits::Block};
 
+const SERVER_ERROR: i32 = 1;
+
 /// Pull API
-#[rpc(server)]
+#[rpc(client, server)]
```

Comment:
Whis is this both a client and server API?

Reply:
@Dengjianping please address this on #571 if it doesn't explicitly need to be a client